sensor/kubernetes/fake/networkpolicy.go

```diff
@@ -0,0 +1,109 @@
+package fake
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/stackrox/rox/pkg/concurrency"
+	networkingV1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type networkPolicyToBeManaged struct {
+	workload      NetworkPolicyWorkload
+	networkPolicy *networkingV1.NetworkPolicy
+}
+
+func createMap(entries int) map[string]string {
+	m := make(map[string]string, entries)
+	for i := 0; i < entries; i++ {
+		m[fmt.Sprintf("key-%d", i)] = fmt.Sprintf("value-%d", i)
+	}
+	return m
+}
+
+func (w *WorkloadManager) getNetworkPolicy(workload NetworkPolicyWorkload) *networkPolicyToBeManaged {
+	labels := createMap(workload.NumLabels)
+	namespace, valid := namespacePool.randomElem()
+	if !valid {
+		namespace = "default"
+	}
+	np := &networkingV1.NetworkPolicy{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "NetworkPolicy",
+			APIVersion: "networking.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      randString(),
+			Namespace: namespace,
+			UID:       newUUID(),
+			CreationTimestamp: metav1.Time{
+				Time: time.Now(),
+			},
+			Labels:      labels,
+			Annotations: createRandMap(16, 3),
+		},
+		Spec: networkingV1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Ingress: []networkingV1.NetworkPolicyIngressRule{},
+			Egress:  []networkingV1.NetworkPolicyEgressRule{},
+			PolicyTypes: []networkingV1.PolicyType{
+				networkingV1.PolicyTypeIngress,
+				networkingV1.PolicyTypeEgress,
+			},
+		},
+	}
+	return &networkPolicyToBeManaged{
+		workload:      workload,
+		networkPolicy: np,
+	}
+}
+
+func (w *WorkloadManager) manageNetworkPolicy(ctx context.Context, resources *networkPolicyToBeManaged) {
+	w.manageNetworkPolicyLifecycle(ctx, resources)
+
+	for count := 0; resources.workload.NumLifecycles == 0 || count < resources.workload.NumLifecycles; count++ {
+		np := w.getNetworkPolicy(resources.workload)
+		if _, err := w.client.Kubernetes().NetworkingV1().NetworkPolicies(np.networkPolicy.Namespace).Create(ctx, np.networkPolicy, metav1.CreateOptions{}); err != nil {
+			log.Errorf("error creating networkPolicy: %v", err)
+		}
+		w.manageNetworkPolicyLifecycle(ctx, resources)
+	}
+}
+
+func (w *WorkloadManager) manageNetworkPolicyLifecycle(ctx context.Context, resources *networkPolicyToBeManaged) {
+	timer := newTimerWithJitter(resources.workload.LifecycleDuration/2 + time.Duration(rand.Int63n(int64(resources.workload.LifecycleDuration))))
+	defer timer.Stop()
+
+	npNextUpdate := calculateDurationWithJitter(resources.workload.UpdateInterval)
+
+	np := resources.networkPolicy
+
+	stopSig := concurrency.NewSignal()
+
+	npClient := w.client.Kubernetes().NetworkingV1().NetworkPolicies(np.Namespace)
```

Comment:
Maybe the empty lines could be reduced here?

Reply:
Will do 👍